draft-ietf-quic-transport.md

```diff
@@ -2721,6 +2721,66 @@ between endpoints.  Application protocols SHOULD define rules for handling
 streams that are prematurely cancelled by either endpoint.
 
 
+# Alternative Initial Set {#alternative-initial}
+
+In order to avoid ossification of the cleartext and obfuscated fields of QUIC
+packets, a server can announce an alternative set of initial values to be used,
+which is comprised of:
+
+* Alternative version number; a 32-bit unsigned number that is to be presented
+  on wire in place of the version number specified in this document.  This value
+  MUST NOT be a reserved version ({{versions}}).
+
+* Packet type modifier; a two-bit value that is to be applied as a bit-wise
+  exclusive or (XOR) to the most significant bits of the Initial, Handshake,
+  0-RTT, and Retry packets. This XOR is applied after the packets are encrypted
+  and before they are decrypted.
```

Comment:
So... This isn't totally clear.  There are two "encryption" or "decryption" operations that might apply: the AEAD for packet protection and the masking for header protection.

I think that you mean to say that the value you feed into the AAD for packet protection is unmodified and this alteration only applies afterwards.

However, I would prefer it if were applied to the raw values so that the XOR is also authenticated by packet protection.  Then you have a clear order of operations for applying this information that matches the order of fields:

0. encode the packet
1. replace the version
2. mask the type (if long header)
3. apply packet and header protection using the new salt

Reply:
Thanks for arguing for that. I had somehow thought that that's impossible, but I do not know why. Addressed in 14469be.